src/client.ts

```diff
@@ -0,0 +1,64 @@
+'use strict';
+import { Strategy } from './strategy';
+import { FeatureInterface } from './feature';
+import Repository from './repository';
+
+export default class UnleashClient {
+    private repository: Repository;
+    private strategies: Strategy[];
+    private errorHandler: (error: Error) => void;
+
+    constructor (repository: Repository, strategies: Strategy[], errorHandler) {
+        this.repository     = repository;
+        this.errorHandler   = errorHandler || function(){};
+        this.strategies     = strategies || [];
+
+        strategies.forEach((strategy: Strategy) => {
+            if (!strategy ||
+                !strategy.name ||
+                typeof strategy.name !== 'string' ||
+                !strategy.isEnabled ||
+                typeof strategy.isEnabled !== 'function'
+            ) {
+                throw new Error('Invalid strategy data / interface');
+            }
+        });
+    }
+
+    private getStrategy (name: string) : Strategy {
+        let match;
+        this.strategies.some((strategy: Strategy) : boolean => {
+            if (strategy.name === name) {
+                match = strategy;
+                return true;
+            }
+            return false;
+        });
+        return match;
+    }
+
+    isEnabled (name: string, context: any) : boolean {
```

Comment:
In Java it's easy to do method overloading, easily making it default to false. Not sure if it is worth the effort in node? It would be a bit tedious checking if it context or defaultValue or both or both is set.